source/game/system/map/MapdataArea.cc

```diff
@@ -0,0 +1,178 @@
+        return nullptr;
+    }
+
+    return courseMap->getPointInfo(m_railId);
+}
+
+MapdataAreaBox::MapdataAreaBox(const SData *data, s16 index) : MapdataAreaBase(data, index) {
+    m_dimensions.x = 0.5f * (10000.0f * m_scale.x);
+    m_dimensions.y = 10000.0f * m_scale.y;
+    m_dimensions.z = 0.5f * (10000.0f * m_scale.z);
+
+    m_ellipseAspectRatio = 0.0f;
+    m_ellipseRadiusSq = 0.0f;
+    m_sqBoundingSphereRadius = m_dimensions.squaredLength();
+
+    EGG::Quatf rotation = EGG::Quatf::FromRPY(DEG2RAD * m_rotation.x, DEG2RAD * m_rotation.y,
+            DEG2RAD * m_rotation.z);
+
+    m_right = rotation.rotateVector(EGG::Vector3f::ex);
+    m_up = rotation.rotateVector(EGG::Vector3f::ey);
+    m_forward = rotation.rotateVector(EGG::Vector3f::ez);
+}
+
+bool MapdataAreaBox::testImpl(const EGG::Vector3f &pos) const {
+    EGG::Vector3f relPos = pos - m_position;
+
+    f32 u = relPos.dot(m_up);
+    if (u > m_dimensions.y || u < 0.0f) {
+        return false;
+    }
+
+    f32 r = relPos.dot(m_right);
+    if (r > m_dimensions.x || r < -m_dimensions.x) {
+        return false;
+    }
+
+    f32 f = relPos.dot(m_forward);
+    if (f > m_dimensions.z || f < -m_dimensions.z) {
+        return false;
+    }
+
+    return true;
+}
+
+MapdataAreaCylinder::MapdataAreaCylinder(const SData *data, s16 index)
+    : MapdataAreaBase(data, index) {
+    m_dimensions = m_scale * 5000.0f;
+    m_ellipseRadiusSq = m_dimensions.x * m_dimensions.x;
+    m_sqBoundingSphereRadius = m_dimensions.x * m_dimensions.x + m_dimensions.z * m_dimensions.z;
+    m_ellipseAspectRatio = m_scale.x / m_scale.z;
+
+    EGG::Quatf rotation = EGG::Quatf::FromRPY(DEG2RAD * m_rotation.x, DEG2RAD * m_rotation.y,
+            DEG2RAD * m_rotation.z);
+
+    m_right = rotation.rotateVector(EGG::Vector3f::ex);
+    m_up = rotation.rotateVector(EGG::Vector3f::ey);
+    m_forward = rotation.rotateVector(EGG::Vector3f::ez);
+}
+
+bool MapdataAreaCylinder::testImpl(const EGG::Vector3f &pos) const {
+    EGG::Vector3f relPos = pos - m_position;
+
+    if (relPos.dot(m_up) > m_dimensions.y) {
+        return false;
+    }
+
+    f32 f = m_ellipseAspectRatio * relPos.dot(m_forward);
+    f32 r = relPos.dot(m_right);
+    if (r * r + f * f < m_ellipseRadiusSq) {
+        return false;
+    }
+
+    return true;
+}
+
+MapdataAreaAccessor::MapdataAreaAccessor(const MapSectionHeader *header)
+    : MapdataAccessorBase<MapdataAreaBase, MapdataAreaBase::SData>(header) {
+    init(reinterpret_cast<const MapdataAreaBase::SData *>(m_sectionHeader + 1),
+            parse<u16>(m_sectionHeader->count));
+
+    m_sortedEntries =
+            std::span<MapdataAreaBase *>(new MapdataAreaBase *[m_entryCount], m_entryCount);
+}
+
+MapdataAreaAccessor::~MapdataAreaAccessor() {
+    delete m_sortedEntries.data();
+}
+
+void MapdataAreaAccessor::init(const MapdataAreaBase::SData *start, u16 count) {
+    if (count != 0) {
+        m_entryCount = count;
+        m_entries = new MapdataAreaBase *[count];
```

Comment:
This leaks

Reply:
Are you sure? MapdataAreaAccessor inherits MapdataAccessorBase, and the dtor of MapdataAccessorBase deletes `m_entries`. The only thing we put in our overridden dtor is the `m_sortedEntries` span. I don't see the leak.